x-pack/plugins/security_solution/public/management/pages/policy/view/ingest_manager_integration/endpoint_package_custom_extension/components/fleet_trusted_apps_card.tsx

```diff
@@ -16,19 +16,47 @@ import {
 } from '../../../../../../../../../fleet/public';
 import { useKibana } from '../../../../../../../../../../../src/plugins/kibana_react/public';
 import { getTrustedAppsListPath } from '../../../../../../common/routing';
-import { TrustedAppsListPageRouteState } from '../../../../../../../../common/endpoint/types';
+import {
+  TrustedAppsListPageRouteState,
+  GetExceptionSummaryResponse,
+} from '../../../../../../../../common/endpoint/types';
 import { PLUGIN_ID as FLEET_PLUGIN_ID } from '../../../../../../../../../fleet/common';
 import { MANAGEMENT_APP_ID } from '../../../../../../common/constants';
+import { useToasts } from '../../../../../../../common/lib/kibana';
 import { LinkWithIcon } from './link_with_icon';
-import { TrustedAppItemsSummary } from './trusted_app_items_summary';
+import { ExceptionItemsSummary } from './exception_items_summary';
+import { TrustedAppsHttpService } from '../../../../../trusted_apps/service';
 
 export const FleetTrustedAppsCard = memo<PackageCustomExtensionComponentProps>(({ pkgkey }) => {
   const {
     services: {
       application: { getUrlForApp },
+      http,
     },
-  } = useKibana<{ application: ApplicationStart }>();
+  } = useKibana<CoreStart & { application: ApplicationStart }>();
+  const toasts = useToasts();
+  const [stats, setStats] = useState<GetExceptionSummaryResponse | undefined>();
+  const trustedAppsApi = useMemo(() => new TrustedAppsHttpService(http), [http]);
 
+  useEffect(() => {
+    const fetchStats = async () => {
+      try {
+        const response = await trustedAppsApi.getTrustedAppsSummary();
+        setStats(response);
+      } catch (error) {
+        toasts.addDanger(
```

Comment:
fwiw, there is another toaster a lot of people use where we don't push down the class into the dependencies and if you want to use an error toaster instead of a danger toaster you can with it and it will add any network errors to the "See more" button so that if the user clicks it they can copy and paste the network error into the forums.

Not required to use it, but it's there if you want to:
[x-pack/plugins/security_solution/public/common/hooks/use_app_toasts.ts](https://github.com/elastic/kibana/blob/master/x-pack/plugins/security_solution/public/common/hooks/use_app_toasts.ts#L28)

There's a lot of use cases of it across the application. It does a more in-depth job with error objects when it encounters them and tries to figure them out and adds the "show more" button. It's been very helpful for us on forum questions and issues.

The way this is written, when you encounter `values: { error }` and the translation sees that error object in the message of `"{error}"` I assume it just does a `.toString()` which would give less info compared to using the one mentioned above. Plus we do not have another translation key to add to the stack.

Not required, but if not, I would consider maybe using `addError` instead of `addDanger` here? Seems like most of the places we use `addError` which would make more sense? Unless I'm outdated and we're moving everything to `addDanger` ;-) which maybe we are?

Reply:
That sounds good to me! I'm ok to do this change, maybe @paul-tavares has some thoughts?

Reply:
Ohhhh - this is good to know @FrankHassanabad. I just recently mentioned to our team that I really needed to take some time and look around in our "common" folders to see whats available _(this after I found out about our `class KibanaServices` hook in `public/common/lib/kibana/services`). This `useAppToasts()` sounds like something our team should def. use (@elastic/security-onboarding-and-lifecycle-mgt  FYI ^^)

That being said, what you suggest here might not work out _(I think I tried at one point to use our `useKibana()` hook)_. This particular component here is actually rendered within Fleet's code _(we register a UI extension that load this async and then renders it)_. I also fear that if we do use allot of the "common" stuff from security solution, we might pull "allot" and increase the size of the component's bundle that fleet code would have to download. For the most part, we try to keep these Fleet UI extensions to use most of the built in kibana services.